app/src/main/java/br/com/caelum/casadocodigo/web/AutorWebClient.kt

```diff
@@ -1,30 +1,31 @@
 package br.com.caelum.casadocodigo.web
 
-import android.util.Log
 import br.com.caelum.casadocodigo.model.Autor
 import retrofit2.Call
 import retrofit2.Callback
 import retrofit2.Response
-import retrofit2.Retrofit
-import retrofit2.converter.jackson.JacksonConverterFactory
 import retrofit2.http.Body
 import retrofit2.http.POST
 
-class AutorWebClient(retrofit: Retrofit) {
-    private val autorService = retrofit.create(AutorService::class.java)
+class AutorWebClient(private val autorService: AutorService) {
```

Comment:
Aqui vou esperar para ver as cenas dos próximos capítulos... O código dessa classe ta super genérico e parece aplicável para muitos outros lugares :)

Reply:
justamente, se é pensando em ficar o mais simples possível, só faz sentido abstrair aqui se tiver mais infos que eu for pegar da web além do autor..

sobre o código como um todo, entendo que dá pra deixar mais simples e entendi a sua sugestão naquele encontro..fez sentido..mas eu queria como próximo passo implementar a conversa do banco com a web..por isso deixei com essa complexidade ainda
e eu continuo achando que é dar responsabilidade demais pro ViewModel pra usar ele tanto na camada pra tela quanto pra camada da web..são duas camadas diferentes e continuo batendo na tecla q ele é mais q só um DTO..ele é um objeto q fica atento ao ciclo de vida..levar ele pra vários pontos da aplicação seria quase como ficar levando o entityManager pra vários pontos tbm, a meu ver..

Reply:
Entendo. Acho que nossa divergência talvez seja a importância de levar o objeto para outros lugares. Se você leva ela para outros lugares através de uma interface, nenhum ponto vai saber que é um viewmodel. Não sei se tem alguma restrição técnica, no sentido dele perder os valores em algum momento fora do nosso controle. Por outro lado, podemos ir para frente e ver como vai ficar o resto do código.